deploy/crd.yaml

```diff
@@ -30,6 +30,40 @@ spec:
     kind: Database
     plural: databases
   scope: Namespaced
+  validation:
+    openAPIV3Schema:
+      properties:
+        spec:
+          required:
+            - users
+            - server
+            - serverType
```

Comment:
we are expecting user to give either of `server` or `serverType`. Not sure if we should mandate both of them. Not sure if there is a better way either. :(

Reply:
will make only server mandatory as server is enough to get the database plugin object.